### PR TITLE
Add 1071 FIG search functionality

### PR DESCRIPTION
**Pull request by contolini**
Add client-side search tool to the FIG sidebar and pass it FIG content to be indexed.

The source code for the search tool is available at https://github.com/cfpb/ctrl-f/

## Additions

- A `fig-search.js` file to initialize the search interface + unit tests.
- A `data-search-section` HTML attribute to the FIG content template that marks each section that should be indexed by the search engine.
- A `package.json` file for the FIG to mark the ctrl-f package as a dependency.
- ~~A bajillion cached npm packages used by the ctrl-f package, many of them are dev dependencies and I'll look into purging them.~~ These were taken care of with https://github.com/cfpb/ctrl-f/pull/2.

## How to test this PR

1. Visit the [FIG demo](https://dev4/data-research/small-business-lending/filing-instruction-guide/202x-filing-instructions-guide/) on dev4.
1. Try the search feature in the left-hand sidebar.
1. Run unit tests with `yarn jest fig-search-spec.js`.

## Screenshots

screenshot

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

**Review by wpears — CHANGES_REQUESTED** [review]
I feel a bit queasy about including all the babel/webpack/testing-library/typescript stuff here. I think a lot of this stuff should be moved to devDependencies in the ctrl-f repo, which would then reduce the runtime deps which cf.gov needs to store a copy of.

[Issue here](https://github.com/cfpb/ctrl-f/issues/1)

**Comment by contolini** [comment]
> I feel a bit queasy about including all the babel/webpack/testing-library/typescript stuff here. I think a lot of this stuff should be moved to devDependencies in the ctrl-f repo, which would then reduce the runtime deps which cf.gov needs to store a copy of.
> 
> [Issue here](https://github.com/cfpb/ctrl-f/issues/1)

I'll try moving *everything* to dev dependencies and see how it goes. My gut says it'll be fine because only the compiled JS in the `dist/` dir is consumed by cf.gov.

**Review by wpears — APPROVED** [review]
It's a masterpiece

File: cfgov/unprocessed/apps/filing-instruction-guide/css/main.less [hunk 1]
```diff
@@ -207,3 +208,7 @@
     display: none;
   } );
 }
+
+#ctrl-f-modal h3 {
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/filing-instruction-guide/css/main.less:212]: Does `ctrl-f-modal` ID come in through the `ctrl-f` package? All I'm seeing set in the markup for an ID in this PR is `id="ctrl-f"`.

File: cfgov/unprocessed/css/molecules/global-search.less [hunk 2]
```diff
@@ -111,47 +111,47 @@
...
+    &-form {
+      position: absolute;
+      width: 100%;
+    }
 
-  &[aria-expanded="true"] &-form {
-    display: block;
-  }
+    &.u-invisible {
+      overflow-x: hidden;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/global-search.less:142]: Do we want to be adding to the u-invisible class or should a new utility be defined?

File: cfgov/unprocessed/apps/filing-instruction-guide/css/main.less (hunk 1 above)
Comment (contolini) [reply cfgov/unprocessed/apps/filing-instruction-guide/css/main.less:212]: Yeah it's in the [ctrl-f package](https://github.com/cfpb/ctrl-f/blob/1d17eb0e77c8ebc63c94b769adf666083a2d2878/src/Modal.tsx#L83) and the sizing of the h3 is the only thing cf.gov overrides.

File: cfgov/unprocessed/css/molecules/global-search.less (hunk 2 above)
Comment (contolini) [reply cfgov/unprocessed/css/molecules/global-search.less:142]: Hmmmm I think I mistakenly linted this file and added it to this PR. Its contents haven't been modified (as can be seen when viewing the file with [whitespace changes ignored](https://github.com/cfpb/consumerfinance.gov/pull/7268/files?w=1#diff-2432c2598b33a417af8c309c27551394bbdecd17cfe28ea9960b3cfbf3071c8b)).

**Pull request merged**